### ISSUE TITLE
Accessibility Feedback for Design Intent: Communication Permissions (Notification Preferences)

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#22526]
## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## Must
- The notification icon must be accessibly read as "notifications" so that it can add equivalent context to the string following it. For example, "Notifications On: Email, Text" instead of "[Icon] On: Email, Text." This can be accomplished through sr-only text or an `aria-label`. 

## Should
- Edit, update, and cancel buttons should all have unique aria-labels e.g. "update healthcare notifications" instead of just "update" 
- Screen reader users should understand groupings in the same way sighted users understand groupings. e.g. using `role="group"`. This will ensure that the full context is provided to screen reader users upon tabbing into a new group or exiting one E.g. "Email, checkbox, checked, American legion events, other notifications" vs. "Email, checkbox, checked, American legion events"
- Lists should be used in read state so screen reader users can known how many notification types exist within each group. A `dl` (definition list) may be most semantically appropriate in this context, but an `ul` unordered list may work with more assistive technology.
- _Should_ the "save" option be used, it must be announced to screen reader users via `aria-live`. This may be tricky to implement if there is a timer that causes it to disappear. Alternatively, we can explore removing it in favor of more descriptive content.

## Consider
- Adding more context to the labels on edit. For example, "Notify by Email" instead of just "Email." Providing an action will make it more explicit as to what the check will do. E.g. "Notify by email, checkbox, checked" (Suggested) vs. "Email, checkbox, checked." (Current)
- Placing the edit button vertically at the bottom left of each group on desktop. This may make it more discoverable to users with low vision and make it more clear that it is editing the full group as opposed to just one line.

**Comment by andaleliz** [comment]
Thanks for the feedback @joshkimux and @noahgelman! Here are my comments:

>The notification icon must be accessibly read as "notifications" so that it can add equivalent context to the string following it. For example, "Notifications On: Email, Text" instead of "[Icon] On: Email, Text." This can be accomplished through sr-only text or an aria-label.

This has been removed from the prototype per feedback from the design practice.

>Edit, update, and cancel buttons should all have unique aria-labels e.g. "update healthcare notifications" instead of just "update"
Screen reader users should understand groupings in the same way sighted users understand groupings. e.g. using role="group". This will ensure that the full context is provided to screen reader users upon tabbing into a new group or exiting one E.g. "Email, checkbox, checked, American legion events, other notifications" vs. "Email, checkbox, checked, American legion events"
Lists should be used in read state so screen reader users can known how many notification types exist within each group. A dl (definition list) may be most semantically appropriate in this context, but an ul unordered list may work with more assistive technology.

I've added all of these to the UX pin docs so developers have them when we go to code

>Adding more context to the labels on edit. For example, "Notify by Email" instead of just "Email." Providing an action will make it more explicit as to what the check will do. E.g. "Notify by email, checkbox, checked" (Suggested) vs. "Email, checkbox, checked." (Current)

Prototype has been updated to reflect this, great suggestion!

>Placing the edit button vertically at the bottom left of each group on desktop. This may make it more discoverable to users with low vision and make it more clear that it is editing the full group as opposed to just one line.

This is something I've been thinking about but would need to be implemented across all profile sections as an improvement. I've captured this on my [running list of things to consider](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/profile/future-improvements.md) and will keep you posted on progress!

>Should the "save" option be used, it must be announced to screen reader users via aria-live. This may be tricky to implement if there is a timer that causes it to disappear. Alternatively, we can explore removing it in favor of more descriptive content.

Question on that bit of feedback - do you mean the autosave version? I assume so but want to double check since both do save changes.  I'll capture this feedback in the documentation on the relevant mock.

**Comment by joshkimux** [comment]
@andaleliz yes, that was referring to the autosave version! Thank you!

**Comment by andaleliz** [comment]
Great! All of the feedback has been documented in the prototype.
